crates/turborepo-lib/src/task_graph/visitor.rs

```diff
@@ -177,18 +190,119 @@ impl<'a> Visitor<'a> {
             // hashing so that downstream tasks can count on the hash existing
             //
             // bail if the script doesn't exist
-            let Some(command) = command else { continue };
+            let Some(_command) = command else { continue };
 
             let output_client = self.output_client();
+            let continue_on_error = self.opts.run_opts.continue_on_error;
             let prefix = self.prefix(&info);
             let pretty_prefix = self.color_cache.prefix_with_color(&task_hash, &prefix);
             let ui = self.ui;
+            let manager = self.manager.clone();
+            let package_manager = self.package_graph.package_manager().clone();
+            let workspace_directory = self.repo_root.resolve(workspace_dir);
+            let errors = errors.clone();
+            let task_id_for_display = self.display_task_id(&info);
 
             tasks.push(tokio::spawn(async move {
+                let task_id = info;
                 let _task_cache = task_cache;
                 let mut prefixed_ui =
-                    Self::prefixed_ui(ui, is_github_actions, &output_client, pretty_prefix);
-                prefixed_ui.output(command);
+                    Self::prefixed_ui(ui, is_github_actions, &output_client, pretty_prefix.clone());
+
+                let mut cmd = Command::new(package_manager.to_string());
+                cmd.args(["run", task_id.task()]);
+                cmd.current_dir(workspace_directory.as_path());
+                cmd.stdout(Stdio::piped());
+                cmd.stderr(Stdio::piped());
+
+                // TODO: avoid passing Duration::MAX and instead allow processes to run without
+                // a timeout
+                let mut process = match manager.spawn(cmd, Duration::MAX) {
```

Comment:
The duration here actually indicates the time between requesting a graceful shutdown and when it is forcibly killed. There is currently no mechanism for having tasks themselves time out.